helper/resource/testing.go

```diff
@@ -752,10 +755,47 @@ func ComposeAggregateTestCheckFunc(fs ...TestCheckFunc) TestCheckFunc {
 	}
 }
 
-// TestCheckResourceAttrSet is a TestCheckFunc which ensures a value
-// exists in state for the given name/key combination. It is useful when
-// testing that computed values were set, when it is not possible to
-// know ahead of time what the values will be.
+// TestCheckResourceAttrSet ensures any value exists in the state for the
+// given name and key combination. The opposite of this TestCheckFunc is
+// TestCheckNoResourceAttr. State value checking is only recommended for
+// testing Computed attributes and attribute defaults.
+//
+// Use this as a last resort when a more specific TestCheckFunc cannot be
+// implemented, such as:
+//
+// - TestCheckResourceAttr: Equality checking of non-TypeSet state value.
+// - TestCheckResourceAttrPair: Equality checking of non-TypeSet state value,
+//                              based on another state value.
+// - TestCheckTypeSet*: Equality checking of TypeSet state values.
+// - TestMatchResourceAttr: Regular expression checking of non-TypeSet state
+//                          value.
+// - TestMatchTypeSet*: Regular expression checking on TypeSet state values.
+//
+// For managed resources, the name parameter is combination of the resource
+// type, a period (.), and the name label. The name for the below example
+// configuration would be "myprovider_thing.example".
+//
+//     resource "myprovider_thing" "example" { ... }
+//
+// For data sources, the name parameter is a combination of the keyword "data",
+// a period (.), the data source type, a period (.), and the name label. The
+// name for the below example configuration would be
+// "data.myprovider_thing.example".
+//
+//     data "myprovider_thing" "example" { ... }
+//
+// The key parameter is an attribute path in Terraform CLI 0.11 and earlier
+// "flatmap" syntax. Keys start with the attribute name of a top-level
```

Comment:
Does it make sense to make references to what is now an "ancient" version of Terraform? Do we have a public doc explaining this `"flatmap" syntax`?

Reply:
I would love to reference something, but I'm not sure if there's any terraform.io documentation left on it. 🙁  I'll do some digging, but I'm doubtful. In place of finding that, I'm wondering if [Go testable examples](https://go.dev/blog/examples) might help at least slightly here.

Reply:
The closest thing I can find is: https://www.terraform.io/language/configuration-0-11/interpolation#attributes-of-other-resources, but I think linking to that would honestly be more confusing than anything. The flatmap bits were a detail of the previous state storage mechanism. I've uploaded some TestCheckFunc examples instead.